datamade/committee-oversight
# Dynamically populate the top nav

## Overview

Closes #76.

This PR follows the pattern established in https://github.com/datamade/lisc-cnda/pull/63 to dynamically add nav pages enabled in Wagtail to the top nav. It also conditionally shows admin actions and the login/logout nav items.

<img width="1552" alt="Screen Shot 2019-09-19 at 4 34 40 PM" src="https://user-images.githubusercontent.com/12176173/65282966-d6fc0d80-dafb-11e9-915a-0e60142fde20.png">

## Testing Instructions

- Stop your application containers and remove your Postgis volume: docker-compose down && docker volume rm committee-oversight_committeeoversight-db-data.
- Restart your application containers and load in data as directed in the README. 
- In the Wagtail admin (http://localhost:8000/cms/), log in, then go to Pages > The Committee Oversight Data Project, then click Add a Page. Add a title and some content, then in the Promote tab, check "Show in menus". Publish your new page.
- Find your new page in the page listing and click "Add a child page". Follow the same process as above.
- With your new pages in tow, head back to the main site. Confirm that your parent and child pages are in the nav bar. Also confirm that you have the options to add and view hearings, that your username is displayed, and that you have the option to log out.
- Log out, and confirm that the admin options are no longer visible in the nav.
